addons/web_editor/static/src/js/wysiwyg/widgets/link.js

```diff
@@ -555,13 +567,47 @@ export class Link extends Component {
         this.state.className = this.state.iniClassName
             .replace(allBtnColorPrefixes, ' ')
             .replace(allBtnClassSuffixes, ' ')
-            .replace(allBtnShapes, ' ');
+            .replace(allBtnShapes, ' ')
+            .replace(/(?<!\S)mb-2(?!\S)/, " ");
         this.state.className += ' ' + keptClasses.join(' ');
         // 'o_submit' class will force anchor to be handled as a button in linkdialog.
         if (/(?:s_website_form_send|o_submit)/.test(this.state.className)) {
             this.state.isButton = true;
         }
     }
+    /**
+     * Makes an array of the closest non empty previousSibling and nextSibling.
+     *
+     * @param {HTMLElement} el
+     * @returns {Node[]} [previousSibling, nextSibling]
+     */
+    _getClosestNonEmptySiblings(el) {
+        return ["previous", "next"].map(side => this._getClosestNonEmptySibling(el, side));
+    }
+    /**
+     * Gets the closest previous or nextSibling, excluding empty text nodes and
+     * <br> tags.
+     *
+     * @param {HTMLElement} el
+     * @param {"previous"|"next"} side - side to check for the sibling
+     * @returns {Node|null}
+     */
+    _getClosestNonEmptySibling(el, side) {
+        const sibling = el[side + "Sibling"];
+        if (sibling) {
+            const isTextNodeEmpty = sibling.nodeType === 3 && !!sibling.textContent.match(/^\s*$/);
+            let isBrNode = false;
+            let isInvisible = false;
+            if (sibling.nodeType === 1) {
+                isBrNode = sibling.nodeName === "BR";
+                isInvisible = !isVisible(sibling);
+            }
+            if (isTextNodeEmpty || isBrNode || isInvisible) {
+                return this._getClosestNonEmptySibling(sibling, side);
+            }
+        }
+        return sibling;
+    }
```

Comment:
- I would rather use a 'while' loop instead of recursion..
- If `isTextNodeEmpty` is true, there is no need to check `isBrNode` or `isInvisible`. Similarly, if `isBrNode` is true, there is no need to check `isInvisible`.

Reply:
- With a while, I actually don't need a second function anymore. Thanks.
- As I removed the single button check earlier, I only need to get the direct siblings which are buttons.
I refactored the function according to these two points.